### ISSUE TITLE
`no-node-access` rule output false-positive warning after 7.5.0 release

### ISSUE THREAD

**Issue by Simek** [testing-library/eslint-plugin-testing-library#1024]
### Have you read the Troubleshooting section?

Yes

### Plugin version

7.5.0

### ESLint version

9.28.0

### Node.js version

22.15.0

### Bug description

When upgrading the `eslint-plugin-testing-library` to the latest release today I have found out few new ESLint `no-node-access` rule warnings which seems to be a false-positives.

### Steps to reproduce

```jsx
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

it('outputs false-positive warnings', async () => {
  render(<button>Test</button>);

  const user = userEvent.setup();
  await user.click(screen.getByText('Test'));
});
```

### Error output/screenshots

![Image](https://github.com/user-attachments/assets/c789257a-7a6e-44ec-963a-4ee714515d86)

### ESLint configuration

ESLint flat config chunk related to Testing Library:

```js
  {
    files: ['**/*-test.[jt]s?(x)'],
    ...testingLibrary.configs['flat/react'],
  }
```

### Rule(s) affected

no-node-access

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
@Simek thanks for reporting.

@y-hsgw do you think you can handle this one? I think just adding `"user"` to `EVENTS_SIMULATORS` should be enough.

**Comment by Simek** [comment]
What if I name the constant differently? I think it's a bit too much to expect that everyone will call it the same. 

Can we instead validate the typing on which the event trigger is called? In this case it would be `UserEvent` coming from `@testing-library/user-event` package.

**Comment by G-Rath** [comment]
> Can we instead validate the typing on which the event trigger is called?

That would make TypeScript a requirement for this to be usable, which not everyone uses (though an enhanced version of this rule for TypeScript-based projects would be a cool addition)

You could improve accuracy by using logic similar to what we do in [`eslint-plugin-jest` with our `parseJestFnCall` utility](
https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/utils/parseJestFnCall.ts), which reasonably accurately resolves Jest functions in vanilla JS contexts - it works by taking a call expression and attempts to resolve it to where it comes "into" the file.

i.e. 

```
import { test as somethingCompletelyDifferent } from '@jest/globals`;

const oneAlias = somethingCompletelyDifferent;
const twoAlias = oneAlias;
const threeAlias = twoAlias;

describe('a weird number of aliases', () => {
  threeAlias('this is a test', () => {
    // ...
  });
});
```

Our utility will correctly determine that both `describe` and `threeAlias` are the Jest `describe` and `test` methods respectively, which is very similar to what you want to do for this except you don't need to handle globals and you'd want to look for `xyz.setup()` in the middle (so you'd first trace a variable back to the `setup` call, then trace the subject of the call to see if it was imported from `@testing-library/user-event`)

There are other improvements that could be made afterwards too, for example while it'd be rare, you could easily also check if the subject has an explicit `UserEvent` type on its declaration (since that's part of the AST if the consumer is using `@typescript-eslint/parser`, so does not require the TypeScript service to leverage)

**Comment by G-Rath** [comment]
I would say though in the meantime it would be good to get a patch doing what @Belco90 suggested since right now this is flagging very valid code (I've got 194 errors in one project alone, just by updating - all of which are false positives 😅)

**Comment by Belco90** [comment]
> What if I name the constant differently? I think it's a bit too much to expect that everyone will call it the same.

That's the ideal fix indeed, but wanted to exclude `user` in the meantime to handle the expected name at least.

@G-Rath thanks for sharing how is handled in `eslint-plugin-jest`! There are other parts of this plugin where we should be doing the same, so it's not just a matter of updating the `no-node-access` rule but detecting renames/remappings in Testing Library in general.

**Comment by dennisoelkers** [comment]
I am seeing similar false positives when using `react-select-event`, like [here](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-web-interface/src/views/logic/valueactions/SelectExtractorType.test.tsx#L65). How can I fix this?

**Comment by john-harding** [comment]
I'm still seeing this in `7.5.2` from this line of code:
```
await user.setup({ pointerEventsCheck: PointerEventsCheckLevel.Never }).click(submitButton);
```

**Comment by yellowled** [comment]
Also getting a lot of false positives in `7.5.3`, our usual setup is something like this:

```js
const userEvt = userEvent.setup();

describe("<Component />", () => {
    it("should test something", async () => {
        await userEvt.click(screen.getByRole("button")); // this triggers a testing-library/no-node-access error
    });
});
```
Also, we can't call it `user` since we pass a `user` to a lot of our components, which in turn gives us `is already declared in the upper scope`. 😅

**Comment by WikiRik** [comment]
@Belco90 is it useful to track the suggestions by @G-Rath in a different issue so it's not forgotten?

**Comment by pstachula-dev** [comment]
Hello! In 7.5.4 I found new issue when I use `fireEvent` from `@testing-library`

In 7.5.3 everything is ok. @Belco90 

<img width="467" height="92" alt="Image" src="https://github.com/user-attachments/assets/0181fec5-6f5a-4a9f-9060-5b4360f8404b" />

**Comment by Belco90** [comment]
> Hello! In 7.5.4 I found new issue when I use `fireEvent` from `@testing-library`
> 
> In 7.5.3 everything is ok. @Belco90 
> 
> <img width="467" height="92" alt="Image" src="https://github.com/user-attachments/assets/0181fec5-6f5a-4a9f-9060-5b4360f8404b" />

Can you report this in a new issue? Thanks.

**Comment by pstachula-dev** [comment]
Added: https://github.com/testing-library/eslint-plugin-testing-library/issues/1038
